src/Sites/app/Runtime/TabCompleteDetection.tsx

```diff
@@ -194,6 +188,86 @@ export class TabCompleteDetection {
 			this.inputValue.next({ message: newMessage, cursorPosition: newCursorPosition });
 		}, 0);
 	}
+
+	/**
+	 * Removes event listeners from the currentInput.
+	 * Finds and sets the new currentInput by querying the DOM.
+	 * Adds event listeners to the new currentInput.
+	 */
+	private updateInput():void{
```

Comment:
All the methods here could be moved to a more general place, such as InputManager. Tab completion is only one of the elements that becomes broken after being timed out

Reply:
The emote menu button needs to be reinjected. Does the "allow sending same message twice" and ctrl + enter also break?